### ISSUE TITLE
Object is possibly 'null' not reported for an if expression with an await inside it

### ISSUE THREAD

**Issue by vjeranc** [microsoft/TypeScript#45046]
# Bug Report

### 🔎 Search Terms

missing warning for object being possibly null

### 🕗 Version & Regression Information

4.3.5 + nightly

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about this and didn't find anything.
### ⏯ Playground Link
[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.4.0-beta#code/MYGwhgzhAEBKYDsAm0DeAoaXoBcBc0CArgLYBGApgE7QA+hRII0AvA0wNzqbYAWkraJACeCYNAAUASlYA+QhQDu0AApUA9iQCWEChJot5unABUtJCuqI4J0udCoSAZmBC6pAGmgBGAAz+pKS4eLCoiBEERMUkZQzQQ7CwtJ0kcXh0AOhxoADIc6ABCCTBFMC1stMz+CGlA+MSG6GB1BAh1EAoMkHUAcwkAcnBwfq9KiCyghMaqChwiKgixrOgAah8AJi5GrABfKewZuYXcdPGcAH5z31WfAGYt7D30J+bW7IMFZXhkaS4qZbYfl8wSi4ic4WAOC0LQc4TsGCwrwg2QADoJ-mEEL8eP9smxiJxntxMb8gA)

### 💻 Code

```ts
class Rand {
    t: number | null = null;

    has = async () => new Promise(r => setTimeout(() => r(false), 1000));

    run = async () => {
        if (this.t && !(await this.has())) {
          console.log('should not be null', this.t);
          return this.t /* this can be null */ + 12;
        }
        return this.t??0 + 13;
    }

}
```

### 🙁 Actual behavior

Compiler reports that code is fine.

### 🙂 Expected behavior

Compiler should report use of `this.t` inside the `if` block as a possible null because `await` call in the `if` condition allows something else to set `this.t` to `null`.

**Comment by RyanCavanaugh** [comment]
See #9998
